finat/direct_serendipity.py

```diff
@@ -0,0 +1,371 @@
+import numpy
+
+from finat.finiteelementbase import FiniteElementBase
+from finat.physically_mapped import DirectlyDefinedElement
+from FIAT.reference_element import UFCQuadrilateral
+from FIAT.polynomial_set import mis
+
+import gem
+import sympy
+from finat.sympy2gem import sympy2gem
+
+
+class DirectSerendipity(DirectlyDefinedElement, FiniteElementBase):
+    def __init__(self, cell, degree):
+        assert isinstance(cell, UFCQuadrilateral)
```

Comment:
Can these be extended to higher dimensions, it's just you didn't do it? If so, do

```
if not isinstance(cell, UFCQuadrilateral):
    raise NotImplementedError(f"Not yet done for {cell} cells, please get in touch")
```